### ISSUE TITLE
tensorflow pb export where are the classes?

### ISSUE THREAD

**Issue by grexai** [ultralytics/yolov5#11646]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,
I would like to create a tensorflow export for my model. It went well and actually  i can retrieve the detections but i cannot find the classes.
I have trained a well-working segmentation modell. Exported it with the following params:
python export.py --weights path/to/mymodel/weights/best.pt --nms --include pb --img 1024
The
After adding combined non max suppression I have four output identity tensors. The first contains the bbs with (100,4) dims which are fine and i validated the detections, then 2 other identities which are probably the scores, however hard to decide which is which. The last one is an integer per image.  Okay the other outputs belongs to the proto masks so its irrelevant in this case. Has anyone succeded to get the classes from tf model

Thanks in advance
![image](https://github.com/ultralytics/yolov5/assets/3353622/fed3d710-3d12-4cd4-8060-331ecfa619d7)

**Comment by glenn-jocher** [comment]
@grexai hello,

Thank you for using YOLOv5! It seems that you're facing an issue getting the classes from the tensorflow model after exporting. Unfortunately, we cannot help you with this as it goes beyond the scope of YOLOv5. However, we highly encourage you to take a look at the TensorFlow documentation and API reference for more information and support.

Best of luck with your project!

**Comment by grexai** [comment]
Thank you for the quick answer.

Well, I have checked the documentation, and based on it the 3rd tensor should be the right tensor to get the classes however these are full of floating point values that are smaller than 1. So it is unclear how they should be the classes and this is the reason why I think it is yolo related.
![image](https://github.com/ultralytics/yolov5/assets/3353622/dd6253d8-8042-4922-ac49-65573f20516e)

**Comment by glenn-jocher** [comment]
@grexai you're welcome, glad to help. It looks like you've tried to use the 3rd tensor to get the classes, which according to the documentation should contain integers representing the class labels. However, you're seeing floating point values smaller than 1 instead. 

While it's difficult to say definitively without more information, one possibility is that during export, the conversion from PyTorch to TensorFlow may have caused some unexpected behavior for this particular model. It's also possible that the issue is unrelated to YOLOv5 specifically. 

To investigate further, we recommend verifying your model's behavior with an alternative setup, such as TensorFlow Serving or the TensorFlow runtime. You may also want to explore TensorFlow documentation and support resources to see if there are additional insights or solutions that can be applied.

**Comment by grexai** [comment]
The 3rd one is a floating point tensor not an integer. As a torch model the behaviour is fine.

**Comment by glenn-jocher** [comment]
@grexai thank you for your response. Based on your experience, it seems that the issue is related to the conversion from PyTorch to TensorFlow. As a PyTorch model, it behaves correctly, but the exported TensorFlow model is producing a floating-point tensor instead of integers.

This suggests that the issue is likely a result of the conversion process, rather than anything specific to YOLOv5. We recommend exploring alternative conversion methods or reaching out to the TensorFlow community for support in diagnosing the issue.

Best,

**Comment by lucastack** [comment]
Hello @grexai, are you using the `detect.py` script? 

Best regards!

**Comment by glenn-jocher** [comment]
Hello @lucastack,

Thank you for reaching out to us. It looks like you're trying to get the class predictions from a TensorFlow exported model, but encountering unexpected floating point values instead of the expected integer class labels. 

Can you please provide more information about the steps you're taking to export and use the model? It would also be helpful if you could share a code snippet of how you're accessing the output tensors to help us better understand the issue.

Best,
